src/server/services/model-version.service.ts

```diff
@@ -627,3 +629,89 @@ export const modelVersionGeneratedImagesOnTimeframe = async ({
+`,
+  [CommercialUse.Sell]: `
+<b>Sale of the Model: Do not sell or license the Model in exchange for a for a fee or something else of value.</b>
+`,
+};
+
+export function addAdditionalLicensePermissions(
+  license: string,
+  options: {
+    modelId: number;
+    modelName: string;
+    versionId: number;
+    username: string | null;
+    allowNoCredit: boolean;
+    allowDerivatives: boolean;
+    allowDifferentLicense: boolean;
+    allowCommercialUse: CommercialUse[];
+  }
+) {
+  license += `
+<b>Attachment B
+
+Additional Restrictions
+
+This Attachment B supplements the license to which it is attached (“License”). In addition to any restrictions set forth in the License, the following additional terms apply.  The below restrictions apply to the Model and Derivatives of the Model, even though only the Model is referenced.  “Merge” means, with respect to the Model, combining the Model or a Derivative of the Model with one or more other models to produce a single model. A Derivative of the Model will be understood to include Merges.
+
+You agree to the following with respect to the Model (each a “Permission”):
+</b>
+`;
+
+  if (!options.allowNoCredit) {
+    const modelUrl = `${env.NEXTAUTH_URL}/models/${options.modelId}?modelVersionId=${options.versionId}`;
+    const creatorUrl = `${env.NEXTAUTH_URL}/user/${options.username}`;
+    const licenseUrl = `${env.NEXTAUTH_URL}/models/license/${options.versionId}`;
+
+    license += `
+<b>Creator Credit: to use or distribute the Model you must credit the creator as follows:
+- Model: ${options.modelName} [${modelUrl}](${modelUrl})
+- Creator: ${options.username} [${creatorUrl}](${creatorUrl})
+- License: [${licenseUrl}](${licenseUrl})
+</b>
+`;
+  }
+
+  if (!options.allowDerivatives) {
+    license += `
+<b>Do not Share or make available a Merge. The following are a few examples of what is prohibited:
+<ol type="a">
+<li>Running an image generation service that uses a Merge; and</li>
+<li>Making a Merge available for deployment by another person on an as-a-service basis, download, or otherwise.</li>
+</ol></b>
+`;
+  }
+
+  if (options.allowDifferentLicense) {
+    license += `
+<b>Changing Permissions: modify or eliminate any of the applicable Permissions when sharing or making available a Derivative of the Model.</b>
+`;
+  }
+
+  if (!options.allowCommercialUse.includes(CommercialUse.Image))
+    license += commercialUsePermissionContent[CommercialUse.Image];
+  if (!options.allowCommercialUse.includes(CommercialUse.Rent))
+    license += commercialUsePermissionContent[CommercialUse.Rent];
+  if (!options.allowCommercialUse.includes(CommercialUse.RentCivit))
+    license += commercialUsePermissionContent[CommercialUse.RentCivit];
+  if (!options.allowCommercialUse.includes(CommercialUse.Sell))
+    license += commercialUsePermissionContent[CommercialUse.Sell];
```

Comment:
Just an opinion here but I think this works better as an array in which you push strings to and then join 🤔. Don't mind it too much, I just think it's better than adding up the string until the end.